### DISCUSSION TITLE
CustomTk Image in frame

### DISCUSSION THREAD

**Discussion by MaxSSD** [TomSchimansky/CustomTkinter#1132] [General]
Hey hey I made a GUI to interact with openai library. Is it possible to display the picture in Frame or label in customtkinter? 

I tried with CTkImage but I don't quite get that the dark and light mode differentiation there. I was close to success with python-resize-image and creating a separate frame and label to StringVar the png but not quite.

Preview:
![Capture7](https://user-images.githubusercontent.com/86234226/215331554-1d9b9623-d958-4a52-87ac-8550cd280c58.PNG)

**Comment by RobbieKhan** [comment]
```
from PIL import Image
import customtkinter
import os

IMAGE_WIDTH = desired_width
IMAGE_HEIGHT = desired_height
IMAGE_PATH = 'image/image.png'

...

your_image = customtkinter.CTkImage(light_image=Image.open(os.path.join(IMAGE_PATH)), size=(IMAGE_WIDTH , IMAGE_HEIGHT))
label = customtkinter.CTkLabel(master=your_master, image=your_image, text='')
label.grid(column=0, row=0)
```

You can just put that label to the desired frame then

**Reply by MaxSSD** [reply]
I just tried the code. It may work in cases where a function doesn't try to open the image in Label/Frame but just set static path of the png. I am not certain though maybe one could set lists to width, height and image and then call them from a function. 

My image goes from URL to string and is then shown with modified parameters adhering to combos/sliders. I tried to get the image path as stringvar to create an image, I think there were some conflicts between customtkinter and ttk buttons.

I tried something like `img.show(self.Photo_var)` you can see in this repository [https://github.com/MaxSSD/customtk-openai/blob/main/openaictk.py](https://github.com/MaxSSD/customtk-openai/blob/main/openaictk.py)

This is the example function that would show pictures in the frame.

skelly:
```python
def dalle(self):
        tk.messagebox.showinfo('Creating and saving image...')
        prompt_ins = self.dalle_prompt.get()
        response = openai.Image.create(
            prompt=prompt_ins,
            n=1,
            size=self.set_picture_size_var.get()
        )
        image_url = response['data'][0]['url']
        webUrl = urllib.request.urlopen(image_url)
        img = Image.open(webUrl)
        file_name = os.path.basename(prompt_ins)[:255] + '.png'
        img.show()
        img.save(file_name)

```

**Reply by RobbieKhan** [reply]
![image](https://user-images.githubusercontent.com/53120051/215699516-66a03eb4-5685-49ca-ad68-1b55968d793f.png)
You should provide `self.img` to the label, not `img`

```
self.img = ImageTk.PhotoImage(img)
self.image_label.config(image=self.img)
```

**Reply by MaxSSD** [reply]
Haha very fair, the error is similar, the function did not quite get there. Something is wrong with resize_cover(). I will try to find new ways of calling resizable images.

I modified it to look like this:
```python
def generate_image(self):
        tk.messagebox.showinfo('Creating and saving image...')
        prompt_ins = self.dalle_prompt.get()
        response = openai.Image.create(
            prompt=prompt_ins,
            n=1,
            size=self.set_picture_size_var.get()
        )
        image_url = response['data'][0]['url']
        webUrl = urllib.request.urlopen(image_url)
        img = Image.open(webUrl)
        file_name = os.path.basename('DallE')[:255] + '.png'
        img.save(file_name)
        # DISPLAY IMAGE IN LABEL OF THE FRAME
        img.show(str(self.image_var.get()))
        img = resizeimage.resize_cover(
            file_name, size=any)
        self.img = ImageTk.PhotoImage(img)
        self.image_label.config(image=self.img)
```

Throws error:
```error
Exception in Tkinter callback
Traceback (most recent call last):
  File "D:\Nowy folder\Python\Pythonn\python.exe\build-Python\RIntelligence\testinin.py", line 238, in generate_image
    img = resizeimage.resize_cover(
  File "C:\local-packages\Python39\site-packages\resizeimage\resizeimage.py", line 31, in wrapper
    validator(image, size)
  File "C:\local-packages\Python39\site-packages\resizeimage\resizeimage.py", line 40, in _is_big_enough
    if (size[0] > image.size[0]) and (size[1] > image.size[1]):

TypeError: 'builtin_function_or_method' object is not subscriptable
```
